src/nimbleparser.nim

```diff
@@ -115,20 +121,43 @@ proc parseNimbleFile*(c: NimbleContext; nimbleFile: string; p: Patterns): Requir
         else:
           result.deps.add (createUrlSkipPatterns(u), query)
 
-proc findNimbleFile*(c: var Reporter; dir: string; ambiguous: var bool): string =
-  result = ""
-  var counter = 0
-  for x in walkFiles(dir / "*.nimble"):
-    inc counter
-    if result.len == 0:
-      result = x
-  if counter > 1:
-    ambiguous = true
-    result = ""
-
-#  if counter > 1:
-#    warn c, dir, "cannot determine `.nimble` file; there are multiple to choose from"
-#    result = ""
+proc findNimbleFile*(c: var Reporter, dir: string): Option[string] =
+  ## Search the given directory for a Nimble file.
+  ##
+  ## An error is reported if there are multiple Nimble files 
+  ## which results in an ambiguity.
```

Comment:
You cannot do that though as `dir` might point into an old git commit with an ambig nimble file. We should ignore such a commit, not produce an error.